docs/guides/user-containers/readme.md

```diff
@@ -108,67 +193,222 @@ spec:
 
-The author of a containerized application will work with the administrator to
-define a pod specification template for the container and to create an
-appropriate NnfContainerProfile resource for the container.  The image and tag
-for the user's container will be specified in the profile.
+New NnfContainerProfile resources may be restricted to a specific user ID or group ID . When a
+`data.userID` or `data.groupID` is specified in the profile, only those Workflow resources having a
+matching user ID or group ID will be allowed to use that profile . If the profile specifies both of
+these IDs, then the Workflow resource must match both of them.
 
-New NnfContainerProfile resources may be created by copying one of the provided
-example profiles from the `nnf-system` namespace.  The examples may be found by listing them with `kubectl`:
+## Creating a Container Workflow
 
-```console
-kubectl get nnfcontainerprofiles -n nnf-system
+The user's workflow will specify the name of the `NnfContainerProfile` in a DW directive. If the
+custom profile is named `red-rock-slushy` then it will be specified in the `#DW container` directive
+with the `profile` parameter.
+
+```bash
+#DW container profile=red-rock-slushy  [...]
 ```
 
-### Workflow Job Specification
+Furthermore, to set the container storages for the workflow, storage parameters must also be
+supplied in the workflow. This is done using the `<storage_name>` (see [Container
+Storages](#container-storages)) and setting it to the name of a storage directive that defines an
+NNF filesystem. That storage directive must already exist as part of another workflow (e.g.
+persistent storage) or it can be supplied in the same workflow as the container. For global lustre,
+the `LustreFilesystem` must exist that represents the global lustre filesystem.
+
+In this example, we're creating a GFS2 filesystem to accompany the container directive. We're using
+the `red-rock-slush` profile which contains a non-optional storage called `DW_JOB_local_storage`:
+
+```yaml
+kind: NnfContainerProfile
+metadata:
+  name: red-rock-slushy
+data:
+  storages:
+  - name: DW_JOB_local_storage
+    optional: false
+  template:
+    mpiSpec:
+      ...
+```
 
-The user's workflow will specify the name of the NnfContainerProfile in a DW
-directive.  If the custom profile is named `red-rock-slushy` then it will be
-specified in the "#DW container" directive with the "profile" parameter.
+The resulting container directive looks like this:
 
 ```bash
-#DW container profile=red-rock-slushy  [...]
+#DW jobdw name=my-gfs2 type=gfs2 capacity=100GB"
+#DW container name=my-container profile=red-rock-slushy DW_JOB_local_storage=my-gfs2
+```
+
+Once the workflow progresses, this will create a 100GB GFS2 filesystem that is then mounted into the
+container upon creation. An environment variable called `DW_JOB_local_storage` is made available
+inside of the container and provides the path to the mounted NNF GFS2 filesystem. An application
```

Comment:
Should you talk about how multiple mounts are handled for gfs2 when there is more than one compute node per rabbit allocated to the job?

Reply:
Yes. I've added a section for GFS2 Index Mounts under the Environment Variables sections and placed a note here.